pkg/proxy/proxy.go

```diff
@@ -50,18 +51,90 @@ func NewProxy(maxLoadTestsRun int, kubeClient *kube.Client, specCreator loadTest
 
 // LoadTestStatus defines response structure for status request
 type LoadTestStatus struct {
-	Type            string `json:"type"`
-	DistributedPods int32  `json:"distributedPods"`        // number of distributed pods requested
-	Namespace       string `json:"loadtestName,omitempty"` // namespace created equals the loadtest name
-	Phase           string `json:"phase,omitempty"`        // jmeter loadtest status
-	HasEnvVars      bool   `json:"hasEnvVars"`
-	HasTestData     bool   `json:"hasTestData"`
+	Type            string            `json:"type"`
+	DistributedPods int32             `json:"distributedPods"`        // number of distributed pods requested
+	Namespace       string            `json:"loadtestName,omitempty"` // namespace created equals the loadtest name
+	Phase           string            `json:"phase,omitempty"`        // jmeter loadtest status
+	Tags            map[string]string `json:"tags"`
+	HasEnvVars      bool              `json:"hasEnvVars"`
+	HasTestData     bool              `json:"hasTestData"`
 }
 
 func getLoadTestType(r *http.Request) apisLoadTestV1.LoadTestType {
 	return apisLoadTestV1.LoadTestType(r.FormValue(backendType))
 }
 
+// List lists all the load tests.
+func (p *Proxy) List(w http.ResponseWriter, r *http.Request) {
+	logger := mPkg.GetLogger(r.Context())
+
+	ctx, cancel := context.WithTimeout(r.Context(), loadtest.KubeTimeout)
+	defer cancel()
+
+	opt := kube.ListOptions{}
+
+	// Build phase filter.
+	params := r.URL.Query()
+
+	if phaseParam := params.Get(phase); phaseParam != "" {
+		opt.Phase = &phaseParam
+	}
+
+	// Build tags filter.
+	if tagsString := params.Get(tags); tagsString != "" {
+		pairs := strings.Split(strings.TrimSpace(tagsString), ",")
+		opt.Tags = make(map[string]string, len(pairs))
+
+		for _, pair := range pairs {
+			pair = strings.TrimSpace(pair)
+			if len(pair) < 1 {
+				continue
+			}
+
+			parts := strings.SplitN(pair, ":", 2)
+			label := strings.TrimSpace(parts[0])
+
+			if len(label) == 0 || len(parts) != 2 {
+				continue
+			}
+
+			value := strings.TrimSpace(parts[1])
+
+			if len(value) != 2 {
+				continue
+			}
+
+			opt.Tags[label] = value
+		}
+	}
```

Comment:
Can we extract this into a func please?

Reply:
This one is for parsing the tags to filter so we don't need the validation